src/App.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import './App.css';
+import Header from './components/Header';
+import ItemList from './components/ItemList';
+import Selector from './components/Selector';
+import mockData from './mockData/restaurantList.json';
+import { appState, restaurant } from './utils/interfaces';
+import { parseJson } from './utils/json';
+import { selectorCategory, selectorFilter } from './utils/types';
+import { sortingByCategory, sortingByFilter } from './domain/restaurantSort';
+import { CATEGORY_OPTIONS, FILTER_OPTIONS } from './utils/constants';
+import { localStorageGetItem } from './utils/localStorage';
+import { typePredicates } from './utils/typeCheck';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface Props {}
+
+class App extends React.Component<Props, appState> {
+  constructor(props: Props | Readonly<Props>) {
+    super(props);
+
+    const localStorageSavedList = typePredicates<Array<restaurant>>({
+      data: parseJson(JSON.stringify(localStorageGetItem('restaurantList'))),
+      initialData: [],
+    });
+
+    const restaurantMockDataList = typePredicates<Array<restaurant>>({
+      data: mockData.restaurantList,
+      initialData: [],
+    });
+
+    const wholeList = [...localStorageSavedList, ...restaurantMockDataList];
+    const currentList = sortingByFilter('이름순', wholeList);
+
+    this.state = {
+      category: '전체',
+      filter: '이름순',
+      wholeList,
+      currentList,
+    };
+  }
+
+  categoryOnChange(e: React.ChangeEvent<HTMLSelectElement>) {
+    const { value } = e.target;
+    if (!this.isFilterOptions<selectorCategory>(value, CATEGORY_OPTIONS)) return;
```

Comment:
오히려 이렇게 작성하는게 가독성이 더 좋아보이네요!

```suggestion
    if (!CATEGORY_OPTIONS.includes(value as selectorCategory)) return;
```

내장 메서드의 경우 JavaScript를 아는 개발자라면 대부분 알고, 사용하겠지만 클린이 직접 작성하신 `isFilterOptions`의 경우 어떤 함수인지 모르기 때문에 해당 함수를 찾아보게 되는데요.

결국 `isFilterOptions`가 하는 역할이 제가 위에 변경한 코드와 크게 다르지 않은거 같아요. 함수를 적절하게 분리하는 것은 좋지만 내장 메서드로 처리할 수 있는 부분들까지 분리하게 되면 오히려 코드를 읽을 때 수고스러움이 늘어날 수도 있다고 생각합니다!